### PR TITLE
Add crazy permalink support, resolves #5

### LINKED ISSUE #5: Create a permalinking system for new sponsors
Use @chorn 's idea to create permalinks for sponsors so you can see when they sponsored and at what level. Each page should have a short blurb about the organization.

### PR DESCRIPTION
**Pull request by chorn**
Ok this is a lot of change.

I am not sure it's 100% ready but I am sure we should look at it before I change anything else.

### CODE REVIEW

**Comment by jtesta** [comment]
Just noticed two things right off the bat:

1. /sponsors/ gives a directory listing, instead of a summary.
2. Most of the sponsors there are from last year.

**Comment by chorn** [comment]
The hugo version we've got configured is old, I'll try changing the config and rebuilding.

**Comment by chorn** [comment]
Yeah awesome. Err wait no, not that. If @vladionescu updates the Hugo version in the buildpack it'll do the right thing.

**Comment by jtesta** [comment]
Yesterday I made sponsor logos float-left of their description text.  That made it much more readable.

**Comment by jtesta** [comment]
Why is public/ needed?  Hugo re-generates that every time something changes, no?

**Comment by chorn** [comment]
We need `public` because the buildpack is stupid.

**Comment by chorn** [comment]
If you assign me the issue I'll make a `Procfile` for us that will let us dump `public/` from the repo.

File: themes/cocoa-eh/layouts/partials/head_includes.html [hunk 1]
```diff
@@ -5,7 +5,7 @@
 {{ end }}
 
 <link rel="stylesheet" href="/css/reset.css">
-<link rel="stylesheet" href="/css/main.css">
```
Comment (vladionescu) [inline themes/cocoa-eh/layouts/partials/head_includes.html]: Why? 😢 This just breaks caching for everyone.

File: themes/cocoa-eh/layouts/section/sponsors.html [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+          {{ end }}
+
+      {{ end }}
+
+      <h2 class="sponsor--community">Community</h2>
+
+      <ul>
+        <li><a href="https://tooolroc.org/">TOOOL Rochester</a></li>
+        <li><a href="http://www.interlockroc.org/">Interlock Rochester</a></li>
+        <li><a href="http://www.rochester2600.com/">Rochester 2600</a></li>
```
Comment (vladionescu) [inline themes/cocoa-eh/layouts/section/sponsors.html]: HTTPS links please

**Comment by jtesta** [comment]
I feel out of the loop as to why we need permalinks for the sponsors.  Do we really get that much back in terms of SEO?  At this point, you already invested effort, so I don't mind moving forward... I'm just not clear on what motivated it to begin with.

A couple notes for the Sponsors page:
   * "Our 2018 Sponsors" just clutters up the page.  Its implied that these are our 2018 sponsors, and there aren't any other years listed, sooo...
   * Instead of "Gold", which is vague, lets have it say "Gold Level", etc.
   * Having the logos inset with the descriptions looks much better (see in prod: https://www.bsidesroc.com/sponsors/)

File: themes/cocoa-eh/layouts/section/sponsors.html [hunk 3]
```diff
@@ -0,0 +1,59 @@
+{{ partial "header.html" . }}
+
+<section class="main">
+  <div class="container">
+    <div class="content">
+      <div class="page-heading">Sponsors</div>
+        <p>
+          If you’re interested in sponsoring this year’s BSidesROC event, please visit <a href="/sponsorship-info/">this page</a> for more information.
```
Comment (vladionescu) [inline themes/cocoa-eh/layouts/section/sponsors.html]: Content should not go in theme files, this should go back in `content/sponsors/index.md`.

File: themes/cocoa-eh/layouts/_default/section.html [hunk 4]
```diff
@@ -9,10 +9,8 @@
             <nav id="subnav-nav">
                 ┗
                 <ul>
-                    {{ range .Data.Pages.ByPublishDate }}
-                        {{ if ne .Title $globalTitle }}
-                            {{ partial "li.html" . }}
-                        {{ end }}
+                    {{ if not ( in .Title $globalTitle ) }}
```
Comment (vladionescu) [inline themes/cocoa-eh/layouts/_default/section.html]: This breaks the About page (section) subnav.

File: themes/cocoa-eh/layouts/section/sponsors.html (hunk 3 above)
Comment (chorn) [reply themes/cocoa-eh/layouts/section/sponsors.html]: I agree, and I made it work, but initially the `_index.md` behavior wasn't working the way I thought it was supposed to.

**Comment by chorn** [comment]
@jtesta Initially I had _all_ of the years on there, but it makes the page huge. I figured we could make a past sponsors section that would list them all or something like that.

As to _why_ this was a thing to do, I think that came out of a meeting where I said it'd been helpful for BarcampRoc... The sponsors saw it as part of our value proposition and it was something I could list as what they'd get out of their donation.

File: themes/cocoa-eh/layouts/_default/section.html (hunk 4 above)
Comment (chorn) [reply themes/cocoa-eh/layouts/_default/section.html]: Fixed

**Comment by jtesta** [comment]
CSec is missing.

**Comment by vladionescu** [comment]
💯 looks good to me

**Pull request merged**